File: src/03/insertionSort/insertionSort.py
def insertionSort(array):
    for i in range(1, len(array)):
        preIndex = i - 1
        current = array[i]
        while preIndex >= 0 and array[preIndex] > current:
            array[preIndex+1] = array[preIndex]
            preIndex-=1
        array[preIndex+1] = current
    return array

array = [1, 3, 2, 6, 5, 7, 8, 9, 10, 0]
print(insertionSort(array))

File: src/03/insertionSort/insertionSort.js
function insertionSort(array) {
    const len = array.length;

    let preIndex, current
    for (let i = 1; i < len; ++i) {
        preIndex = i - 1;
        current = array[i];

        while (preIndex >= 0 && array[preIndex] > current) {
            array[preIndex + 1] = array[preIndex]
            preIndex--
        }

        array[preIndex+1] = current
    }

    return array
}

const array = [1, 3, 2, 6, 5, 7, 8, 9, 10, 0]
console.log(insertionSort(array))
